google/field_helpers.go

```diff
@@ -6,8 +6,11 @@ import (
 )
 
 const (
-	globalLinkTemplate    = "projects/%s/global/%s/%s"
-	globalLinkBasePattern = "projects/(.+)/global/%s/(.+)"
+	globalLinkTemplate          = "projects/%s/global/%s/%s"
+	globalLinkBasePattern       = "projects/(.+)/global/%s/(.+)"
```

Comment:
I still don't enjoy seeing a mixture of regexps and formatting patterns (when reading this it's not obvious to me whether the regexp is matched before or after the print). However if this ends up reducing the amount of code needed (which I'm guessing it does given the large number of constants) then I'll defer to your judgement

Alternatively, consider inlining these (especially if they're only used once). Then the formatting/regexp matching is much more explicit

Reply:
`*Template` is used in more than one place.
`*BasePattern` is used in only one place but I like to have both next to each other. It is easy to catch a mistake.